### ISSUE TITLE
Login with OIDC (per Matrix-authentification-server)

### ISSUE THREAD

**Issue by genofire** [vector-im/element-call#2496]
#### What would you like to do?
login on a server, which is setup for element-x-android (and synapse with required additional component: [Matrix-Authentification-Server (MAS)](https://matrix-org.github.io/matrix-authentication-service/) and [Sliding-Sync](https://github.com/matrix-org/sliding-sync)) 

#### Why would you like to do it?
use same server for element-x and element call

#### How would you like to achieve it?
support login per oidc / MAS

**Comment by robintown** [comment]
We would like to ideally remove the ability to log into an account or sign up for one, and just keep the "passwordless guest" mechanism that we currently have. However, we may still want to support OIDC anyways just to make setup with the homeserver as seamless as possible going forward (for homeservers that don't enable legacy auth by default anymore).

**Comment by wrenix** [comment]
So the idea is, that we run two matrix-server?
One for textual communication and one with guestmode for element-call?

---

I believe the guestmode is not a good idea, for secure calls. How could i know, that the user on the otherside is really the "called" one from the element-x client? (if element-call does not verify the user against the homeserver).

Or do i overlook something?

**Comment by fkwp** [comment]
Hi @wrenix,

TL/DR If you want to benefit from Matrix trust anchor we propose to use MatrixRTC / Element Call within your messenger application. In terms of the Element clients that is Element Web/Desktop and the mobile Element X Applications are capable of MatrixRTC using Element Call as a widget.

> One for textual communication and one with guestmode for element-call?

This is a bit more complex. We do have federation in general in Matrix and MatrixRTC is designed on top of Matrix. So MatrixRTC session of course work also in federated rooms. The only reason to use a guest only home server is to apply, e.g., retention rules and have the guest users in a separated user db. This home server is also required to have open registration -- which you really need to take care a lot if used with federation. 

> not verify the user against the homeserver

Users are never verified against a home server. You can trust another users Identity. And the other user can cross sign their devices with their own identity. This way you have a trust-chain.

> I believe the guestmode is not a good idea, for secure calls.

Adding the whole trust concept (cross-signing and going forward invisible crypto) of the regular Matrix Messenger Apps to the Single Page Application (SPA) experience would add big UX complexity making joining a call not a lightweight operation UX wise. Element Call is going forward meant to be used "only" for guest access attending calls rather than creating new ones, where you want to share a call link (from the Messenger App) for people who don't have a regular Matrix account. Those rooms can/should be configured as knocking preventing auto-join guest users. So each guest-user is added by a manual operation -- and if the guest does not match your expectation you can kick them out of the matrix room. Going forward we also want to add an individual token proofing that the guest user was invited from a trusted user.

**Comment by wrenix** [comment]
Thanks for the detailed answer.
After some more research i und erstand the concet now.

The Element Client does really allow Widget to use / handle the Identity.

My Server runs, with everything (together with my single-node livekit). Thank you.

---

PS: The Element-X Clients has an hardcoded Element-Call Website inside? And the just usw the livekit-jwt which is in the well-known oft the Server.

**Comment by fkwp** [comment]
The Element Call widget just adds the MatrixRTC capability to the underlying client. Hence, it's also not too important from which source you download the widget -- since its using the backend component announced from the home servers .well-known. 

> The Element Client does really allow Widget to use / handle the Identity.

The widget does not need to do it... Since its the underlying messenger client which just gets additional features implemented via the widget.

 > The Element-X Clients has an hardcoded Element-Call Website inside?

So far the Element Call widget URL is kind of hardcoded -- remember its just about the MatrixRTC implementation -- not about the config of the backend infrastructure. However, in the recent TestFlight version you can at least config it in the advanced settings. And in the nightlies version the Widget URL can also be configured via .well-known -> see here https://github.com/element-hq/element-meta/issues/2441

**Comment by genofire** [comment]
okay everything is clarified ...
